client/utils/__tests__/api-spec.js

```diff
@@ -0,0 +1,78 @@
+import { expect } from 'chai'
+
+import { withDefaults } from '../api'
+
+describe('withDefaults', () => {
+  it('uses GET by default', () => {
+    const options = withDefaults({})
+
+    expect(options.method).to.eq('GET')
+  })
+
+  it('can override the http method', () => {
+    const options = withDefaults({ method: 'POST' })
+
+    expect(options.method).to.eq('POST')
+  })
+
+  it('adds the content-type header', () => {
+    const options = withDefaults({})
+
+    expect(options.headers).to.contain({ 'Content-Type': 'application/json' })
+  })
+
+  it('JSON stringifies the body', () => {
+    const body = { foo: 42 }
+    const options = withDefaults({ body })
+
+    expect(JSON.parse(options.body)).to.eql(body)
+  })
+
+  it('converts body keys to snake_case', () => {
+    const options = withDefaults({ body: { fooBar: 42 } })
+    const expected = { foo_bar: 42 } // eslint-disable-line camelcase
+
+    expect(JSON.parse(options.body)).to.eql(expected)
+  })
+
+  it('converts the SUCCESS action to a type descriptor', () => {
+    const options = withDefaults({
+      types: ['R', 'S', 'F']
+    })
+
+    expect(options.types[1].type).eql('S')
+  })
+
+  it('keeps SUCCESS and FAILURE actions as type descriptor', () => {
+    const options = withDefaults({
+      types: [
+        'R',
+        {
+          type: 'S',
+          meta: 'SUCCESS'
+        },
+        {
+          type: 'F',
+          meta: 'FAILURE'
+        }
+      ]
+    })
+
+    expect(options.types[1]).to.have.property('meta', 'SUCCESS')
+    expect(options.types[2]).to.have.property('meta', 'FAILURE')
+  })
+
+  it('camelizes payload keys on SUCCESS', () => {
+    const options = withDefaults({
+      types: ['R', 'S', 'F']
+    })
+    const payload = {
+      foo_bar: 42 // eslint-disable-line camelcase
+    }
+    const response = JSON.stringify(payload)
+    const actual = options.types[1].payload(null, null, response)
+    const expected = { fooBar: 42 }
+
+    expect(actual).to.eventually.eql(expected)
+  })
+})
```

Comment:
We need a spec for the FAILURE payload as well.  Not your fault - your source material didn't have one either.  But we should add one here.